.github/workflows/cd.yml

```
@@ -11,43 +11,73 @@ permissions:
-          restore-keys: |
-            ${{ runner.os }}-pnpm-store-
+          context: .
+          push: true
+          tags: ${{ steps.meta-server.outputs.tags }}
+          labels: ${{ steps.meta-server.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
 
-      - name: Install dependencies
-        run: pnpm install --no-frozen-lockfile
+  deploy-staging:
```

actionlint (1.7.10)

[error] job "deploy-staging" needs job "build-and-push" which does not exist in this workflow

(job-needs)